### PR TITLE
set up DB-based user authentication

### PR DESCRIPTION
**Pull request by AnaLanda**

### CODE REVIEW

**Review by KostiantynVorobiov — APPROVED** [review]
Good job!)

File: src/main/java/com/cinema/security/CustomUserDetailsService.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userService.findByEmail(email)
+                .orElseThrow(() -> new UsernameNotFoundException("User was not found"));
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/security/CustomUserDetailsService.java]: it's better to add info about email here

File: src/main/java/com/cinema/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -1,29 +1,48 @@
...
+    public AuthenticationServiceImpl(PasswordEncoder passwordEncoder, UserService userService,
                                      ShoppingCartService shoppingCartService) {
+        this.passwordEncoder = passwordEncoder;
         this.userService = userService;
         this.shoppingCartService = shoppingCartService;
     }
 
     @Override
-    public User register(String email, String password) {
+    public User login(String email, String password) throws AuthenticationException {
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/security/AuthenticationServiceImpl.java]: we don't need login, if we use login form from spring

File: src/main/java/com/cinema/dao/impl/RoleDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl implements RoleDao {
+    private static final Logger log = Logger.getLogger(UserDaoImpl.class);
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/dao/impl/RoleDaoImpl.java]: check this line again😉

File: src/main/java/com/cinema/config/SecurityConfig.java [hunk 4]
```diff
@@ -1,30 +1,44 @@
...
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private UserDetailsService userDetailsService;
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/config/SecurityConfig.java]: why one field is final and second is not?

Comment (AnaLanda) [reply src/main/java/com/cinema/config/SecurityConfig.java]: Because I'm inattentive 🤦‍♀️

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**